source/server/admin/admin.cc

```diff
@@ -615,24 +631,60 @@ ProtobufTypes::MessagePtr AdminImpl::dumpEndpointConfigs() const {
   return endpoint_config_dump;
 }
 
+ProtobufTypes::MessagePtr
+AdminImpl::dumpUnreadyTargetsConfigs(const absl::optional<std::string>& mask) const {
+  auto unready_targets_config_dump_list =
+      std::make_unique<envoy::admin::v3::UnreadyTargetsConfigDumpList>();
+
+  if (mask.has_value()) {
+    const std::string& mask_string = mask.value();
+    if (mask_string == "listener") {
+      dumpListenerUnreadyTargetsConfigs(*unready_targets_config_dump_list);
+    }
+    // More options for unready targets config dump.
+  } else {
+    // Dump all possible information of unready targets.
+    dumpListenerUnreadyTargetsConfigs(*unready_targets_config_dump_list);
+    // More unready targets to add into config dump.
+  }
```

Comment:
The recent update helps to dump everything as @lizan suggested. Therefore, any future need like config dump of filter chain unready targets could be added here.